src/app/shared/components/template/services/template-translate.service.ts

```diff
@@ -127,6 +127,32 @@ export class TemplateTranslateService extends AsyncServiceBase {
     return translated;
   }
 
+  /**
+   * Translate all rows from a data_list. Uses `_translatedFields` metadata to determine which columns
+   * require translation, extracted from names with suffix `::eng`
+   *
+   * NOTE - this method is currently only used by data-items as regular processor translates after parsing
+   * into template, where translateRow method can be used
+   */
+  public translateDataListRows(rows: FlowTypes.Data_listRow[] = []) {
+    if (rows.length === 0) return rows;
+    // use first row to identify list of fields for translation
+    const [{ _translatedFields }] = rows;
+    if (!_translatedFields) return rows;
+    const translateKeys = Object.keys(_translatedFields);
+    // translate rows
+    return rows.map((row) => {
+      for (const key of translateKeys) {
+        const sourceText = row[key];
+        // TODO - confirm whether potential cases that list includes non-text entries
```

Comment:
I can't parse this comment, is the thought that we might want to translate non-text entries, specifically lists? If so, I agree – I think columns named `*_list` will be arrays at this point, so if that's a pattern we want to support then we may need to handle it explicitly.

@esmeetewinkel, do you see using a `*_list` column in a data list to define a list within a single entry as a feature that we currently support (and would want the constituent list entries to be translated)?

Reply:
> [@esmeetewinkel](https://github.com/esmeetewinkel), do you see using a *_list column in a data list to define a list within a single entry as a feature that we currently support (and would want the constituent list entries to be translated)?

I do use `data_list` columns ending in `_list` already (for adding "tags", e.g. [here](https://docs.google.com/spreadsheets/d/1zMzxZ_GF4i9m0EpA3InD8kZSo8vxAQEmHnAKS8ou7Qk/edit?gid=0#gid=0&range=G1)). I've not yet used these for translatable things, but I could imagine that being desired in the future (e.g. specifying the entries of a combo box in a list)

Reply:
Yeah that's basically the sidenote here, I think probably fine to leave unsupported for now and probably over time move away a bit more from the pattern of inline arrays (probably better to handle with data_lists that would have better translation support). If we need support for inline array translation in the future in the future we can open an issue to address